examples/go/cat.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2019, 2020 Aleksa Sarai <cyphar@cyphar.com>
```

Comment:
I'd prefer an alternative example to `cat` but I can do that myself later. But can you please move this to `examples/cat.go`. My plan is to not have separate directories for each language binding (though I might change that in the future).

Reply:
I move it to the examples folder.
If you describe your vision on the go example I will manage that.

Reply:
Oohh an interesting observation. When I move it there it is no longer can be built. If you ran it with `go run` it works but when you run with `go build` it fails with `can't load package: package .: C source files not allowed when not using cgo or SWIG: cat.c` not sure what causes this. When it in a sub directory it is able to be built

Reply:
Right, it's because you're using `cgo` in the bindings and Go has weird rules about what files can be in the same directory.

Okay, keep it in a separate directory for now and I'll deal with it later.